app/routes/signup.js

```diff
@@ -15,6 +18,7 @@ export default class SignupRoute extends Route {
       );
     }
     if (response.status === 200 && !userData.incompleteUserDetails) {
+      mixpanel.track('User Already Registered');
```

Comment:
we should not be using mixpanel every single time, tomorrow if we decide to move away from mixpanel, this makes us job tough. There should a wrapper function which we call from outside and that wrapper function should take care of sending the event to the analytics service (which is mixpanel in this case). Rest of the code doesn't need to know if we're using mixpanel or no. Rest of the code will call just that one function and that function will take it forward.

I hope that makes sense. I had mentioned this in the issue as well #197

Reply:
I read the issue description and comments before started working on this but I couldn't think of a way to have this.
What I thought of and implemented this way, I am mentioning here :
- When a certain code executes suppose we click on submit button in the form and the request is successful then after that execution we will send the details to mixpanel That a new user is logged in/ signed up.
So, I have a question here that suppose if we create a wrapper function and call it from outside, how can we know that the sign-in was successful or not because we don't have any context for what is actually executing in the code, so the best way I could think of is to track that in the code itself. 
Please suggest if you have any other options.

- The wrapper function will be a separate code, so how can that code keep a track of what is executing at the current time and are the code/requests are successfully executed or not.

Reply:
I didn't get your point. We'll be the calling that wrapper function exactly on this line, right?

Reply:
Ohh ok ok, I get your point. Made the changes, please have a look